blockchain/types/transaction.go

```diff
@@ -300,11 +300,14 @@ func (tx *Transaction) EffectiveGasTip(baseFee *big.Int) *big.Int {
 // After hard fork, we need to consider new getEffectiveGasPrice API that operates differently
 // effectiveGasPrice = msg.EffectiveGasPrice(baseFee)
 func (tx *Transaction) EffectiveGasPrice(baseFee *big.Int) *big.Int {
+	if baseFee != nil {
```

Comment:
I think we need to add one more condition to check hard fork.
When baseFee is 0, we need to return gasPrice, not baseFee.

Reply:
Is there a case when `header.BaseFee` is not `nil` and it's value is 0?
cc @kjeom

Reply:
Oh, i confused with rpc return field. never mind this comment! thank you :)